src/translator/cache.h

```diff
@@ -0,0 +1,205 @@
+#ifndef SRC_BERGAMOT_CACHE_H_
+#define SRC_BERGAMOT_CACHE_H_
+
+#include <list>
+#include <mutex>
+#include <unordered_map>
+#include <utility>
+
+#include "3rd_party/L4/inc/L4/LocalMemory/HashTableService.h"
+#include "common/hash.h"
+#include "processed_request_sentence.h"
+#include "translator/beam_search.h"
+
+namespace marian {
+namespace bergamot {
+
+class TranslationModel;
+
+struct CacheKey {
+  const TranslationModel *model;
+  const marian::Words &words;
+};
+
+struct HashCacheKey {
+  size_t operator()(const CacheKey &key) const;
+};
+
+struct CacheConfig {
+  size_t sizeInMB{20};
```

Comment:
Those can't change once the cache is initialised, right? In that case they should be `const`'d. Also, they should be initialised in the constructor. The ones which are not modifiable by the constructor should get `const constexpr` treatment.

Reply:
const is treated at the whole object passing (members where these are kept as a whole are consts, incoming args are consts as a whole).